src/renate/benchmark/datasets/base.py

```diff
@@ -0,0 +1,83 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+from abc import ABC
+from pathlib import Path
+from typing import Optional, Union
+
+from renate import defaults
+from renate.data.data_module import RenateDataModule
+
+
+class TimeIncrementalDataModule(RenateDataModule, ABC):
```

Comment:
They are used for verification in the `TimeIncrementalScenario`. Everything that isn't of this type, is refused. Alternative would be to replace that check with `hasattr("time_step")`. Even though this seems a bit of an overkill to introduce this class simply for a single attribute, I think it's cleaner since this defines the API expected by the respective scenarios.